research/python/src/features/hedge_pressure_research.py

```
@@ -0,0 +1,406 @@
+
+    if not isinstance(spot_price, (int, float)) or spot_price <= 0:
+        raise ValueError("Spot price must be a positive number")
+
+    # Get all strikes and compute totals
+    all_strikes = set(calls_oi.keys()) | set(puts_oi.keys())
+    call_total = sum(calls_oi.values())
+    put_total = sum(puts_oi.values())
+    total_oi = call_total + put_total
+
+    if len(all_strikes) < config.min_strikes:
+        warnings.warn(
```

Ruff (0.14.10)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)